docs/advanced-usage/pipeline.md

```diff
@@ -84,26 +84,27 @@ class SongMetadata
     ) {}
 }
 
-class Song extends Data
+class SongData extends Data
 {
     public function __construct(
         public string $title,
         public SongMetadata $metadata,
     ) {}
     
-    public static function prepareForPipeline(Collection $properties) : Collection
+    public static function prepareForPipeline(array $properties): array
     {
-        $properties->put('metadata', $properties->only(['release_year', 'producer']));
-        
-        return $properties;
+        $collection = collect($properties);
+        $collection->put('metadata', $collection->only(['release_year', 'producer']));
+
+        return $collection->toArray();
```

Comment:
I think @alexrififi did it much better with his [PR](https://github.com/spatie/laravel-data/pull/663/files).
So maybe merge this PR first and then merge his PR, so it override this one.

I know it is much easier to change it here, but I didn't change it here because it felt like stealing 😁

Reply:
Good to know thanks!